crates/ruff_graph/src/resolver.rs

```diff
@@ -0,0 +1,102 @@
+use crate::db::ModuleDb;
+use red_knot_python_semantic::{Module, ModuleName};
+use ruff_db::files::FilePath;
+
+use crate::collector::CollectedImport;
+use ruff_python_ast::name::{QualifiedName, QualifiedNameBuilder};
+use ruff_python_stdlib::identifiers::is_identifier;
+use std::borrow::Cow;
+
+/// Collect all imports for a given Python file.
+pub(crate) struct Resolver<'ast> {
+    module_path: Option<&'ast [String]>,
+    db: &'ast ModuleDb,
+}
```

Comment:
I should have brought this up before. Using the salsa queries directly is something we want to red-knot's core (ideally the fact that we use salsa should be mostly hidden). Could you have a look if you can use `SemanticModel` model here which we consider to be public api